src/Lean/Meta/IndPredBelow.lean

```diff
@@ -279,31 +281,33 @@ where
   closeGoal (vars : BrecOnVariables) (m : MVarId) : MetaM Unit := do
     unless ←isExprMVarAssigned m do
       let m ← introNPRec m
-      unless ←solveByAssumption m do
-        let subGoals ← applyIH m vars
-        subGoals.forM (closeGoal vars)
+      unless ←backwardsChaining m 10 do
+        withMVarContext m do
+        throwError "couldn't solve by backwards chaining: {MessageData.ofGoal m}"
```

Comment:
We could mention the depth threshold in the error message, and how the option name for changing it.

Reply:
improved the message.